includes/Checker/Abstract_Check_Runner.php

```diff
@@ -174,10 +164,18 @@ public function prepare() {
 	 * @return Check_Result An object containing all check results.
 	 */
 	public function run() {
+		global $wpdb, $table_prefix;
 		$checks       = $this->get_checks_to_run();
 		$preparations = $this->get_shared_preparations( $checks );
 		$cleanups     = array();
+		$old_prefix   = null;
 
+		// Set the correct test database prefix if required.
+		if ( $this->has_runtime_check( $checks ) ) {
+			$old_prefix = $wpdb->set_prefix( $table_prefix . 'pc_' );
```

Comment:
Unrelated to this PR, but seeing the logic to set the prefix (and to restore it further below) makes me think it would be good to move this logic into a new `Preparation`, e.g. `Use_Check_Tables_Preparation`. That would be a very small class, but it would be good to put the logic in there.

We can do that separately and it isn't urgent, but would be great to open an issue for it.

Reply:
Thanks @felixarntz I've created a new issue for this here https://github.com/10up/plugin-check/issues/159